src/Logic.hs

```diff
@@ -483,11 +477,65 @@ pushCandidate (pullRequestId, pullRequest) state = do
 -- effects if it does not change the state.
 proceedUntilFixedPoint :: ProjectState -> Action ProjectState
 proceedUntilFixedPoint state = do
-  newState <- proceed state
+  newState <- provideFeedback state >>= proceed
   if newState == state
     then return state
     else proceedUntilFixedPoint newState
 
+-- Describe the status of the pull request.
+describeStatus :: PullRequestId -> PullRequest -> ProjectState -> Text
+describeStatus prId pr state = case Pr.classifyPullRequest pr of
+  PrStatusAwaitingApproval -> "Pull request awaiting approval."
+  PrStatusApproved ->
+    let approver = fromJust $ Pr.approvedBy pr
+    in case Pr.getQueuePosition prId state of
+      0 -> format "Pull request approved by @{}, rebasing now." [approver]
+      1 -> format "Pull request approved by @{}, waiting for rebase at the front of the queue." [approver]
+      n -> format "Pull request approved by @{}, waiting for rebase behind {} pull requests." (approver, n)
+  PrStatusBuildPending ->
+    let Sha sha = fromJust $ getIntegrationSha pr
+    in Text.concat ["Rebased as ", sha, ", waiting for CI …"]
+  PrStatusIntegrated -> "The build succeeded."
+  PrStatusFailedConflict ->
+    let
+      Branch targetBranchName = fromJust $ getIntegrationTargetBranch pr
+      Branch prBranchName = Pr.branch pr
+    in Text.concat
+      [ "Failed to rebase, please rebase manually using\n\n"
+      , "    git rebase --interactive --autosquash origin/"
+      , targetBranchName
+      , " "
+      , prBranchName
+      ]
+  PrStatusFailedBuild -> "The build failed."
+  where
+    getIntegrationSha :: PullRequest -> Maybe Sha
+    getIntegrationSha pullRequest =
+      case Pr.integrationStatus pullRequest of
+        Integrated sha _ -> Just sha
+        _                -> Nothing
+
+    getIntegrationTargetBranch :: PullRequest -> Maybe Branch
+    getIntegrationTargetBranch pullRequest =
+      case Pr.integrationStatus pullRequest of
+        Conflicted targetBranch -> Just targetBranch
+        _                       -> Nothing
+
+-- Leave a comment with the feedback from 'describeStatus' and set the
+-- 'needsFeedback' flag to 'False'.
+feedback :: (PullRequestId, PullRequest) -> ProjectState -> Action ProjectState
```

Comment:
A better name then would be something like `leaveFeedback`, as this also posts the comment.